diff -rupN package/lib/pty.js pty.js-0.1.2/lib/pty.js
--- package/lib/pty.js	2012-05-30 10:21:25.000000000 +0800
+++ pty.js-0.1.2/lib/pty.js	2014-10-03 23:05:57.617741522 +0800
@@ -5,6 +5,7 @@
  */
 
 var net = require('net');
+var tty = require('tty');
 var pty = require('../build/Release/pty.node');
 
 /**
@@ -60,7 +61,7 @@ function Terminal(file, args, opt) {
 
   // fork
   term = pty.fork(file, args, env, cwd, cols, rows);
-  this.socket = new net.Socket(term.fd);
+  this.socket = new tty.ReadStream(term.fd);
   this.socket.setEncoding('utf8');
   this.socket.resume();
 
diff -rupN package/src/pty.cc pty.js-0.1.2/src/pty.cc
--- package/src/pty.cc	2012-05-28 23:30:18.000000000 +0800
+++ pty.js-0.1.2/src/pty.cc	2014-10-03 23:05:57.617741522 +0800
@@ -538,3 +538,5 @@ init(Handle<Object> target) {
   NODE_SET_METHOD(target, "resize", PtyResize);
   NODE_SET_METHOD(target, "process", PtyGetProc);
 }
+
+NODE_MODULE(pty, init);
